### PR TITLE
Reset state machine on final page

### PR DESCRIPTION
**Pull request by alanplatt**

### CODE REVIEW

**Review by Andrew-Folga — DISMISSED** [review]
LGTM barring some missing tests and sonar/pipeline failing

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ffc-grants-eligibility-checker",
   "description": "FFC Grant Eligibility Checker",
-  "version": "1.0.23",
```
Comment (ScenarioFactory) [inline package.json:4]: Needs the package-lock file also?

File: src/config/machines/example-grant-machine.js [hunk 2]
```diff
@@ -23,11 +23,41 @@ export const actionImplementations = {
...
+   * @param {object} context machine context
+   */
+  loadPageAction: (context) => {
+    console.log(`${context.currentPageId} page loaded. State should reset in 1 second.`);
+  },
+
+  resetState: assign({
+    /**
+     * Resets the current page id.
+     * Sets the currentPageId to 'start' and clears the completedPageIds array.
```
Comment (ScenarioFactory) [inline src/config/machines/example-grant-machine.js]: Is this description correct? The function appears to just return the current page ID, no setting or clearing in the function itself?

File: src/config/machines/example-grant-machine.js [hunk 3]
```diff
@@ -23,11 +23,41 @@ export const actionImplementations = {
...
+     * @param {object} _context - The machine context.
+     * @param {object} _event - The triggered event.
+     * @returns {string} The start page id.
+     */
+    currentPageId: (_context, _event) => {
+      return 'start';
+    },
+
+    /**
+     * Resets the completed page ids array
```
Comment (ScenarioFactory) [inline src/config/machines/example-grant-machine.js:48]: Does this reset anything in itself?

File: src/config/machines/example-grant-machine.js (hunk 2 above)
Comment (Andrew-Folga) [reply src/config/machines/example-grant-machine.js]: I have rebased and cleared this up.

File: src/config/machines/example-grant-machine.js (hunk 3 above)
Comment (Andrew-Folga) [reply src/config/machines/example-grant-machine.js:48]: Yes it resets the completed pages ids?

Comment (ScenarioFactory) [reply src/config/machines/example-grant-machine.js:48]: Okay just seems to return an empty array to me but I think I'm missing something

**Review by ScenarioFactory — APPROVED** [review]

**Pull request merged**